src/detector/detector.py

```diff
@@ -143,171 +100,33 @@ def supported_request_types(self) -> dict[type[MCastRequest], Callable[[dict], M
-
-    def get_capture_image(self, **kwargs) -> GetCaptureImageResponse:
-        """
-        :key request: GetCaptureImageRequest
-        """
-
-        request: GetCaptureImageRequest = get_kwarg(
-            kwargs=kwargs,
-            key="request",
-            arg_type=GetCaptureImageRequest)
-
-        encoded_frame: bool
-        encoded_image_rgb_single_row: numpy.array
-        encoded, encoded_image_rgb_single_row = cv2.imencode(request.format, self._captured_image)
-        encoded_image_rgb_bytes: bytes = encoded_image_rgb_single_row.tobytes()
-        encoded_image_rgb_base64 = base64.b64encode(encoded_image_rgb_bytes)
-        return GetCaptureImageResponse(
-            format=request.format,
-            image_base64=encoded_image_rgb_base64)
-
-        # img_bytes = base64.b64decode(img_str)
-        # img_buffer = numpy.frombuffer(img_bytes, dtype=numpy.uint8)
-        # img = cv2.imdecode(img_buffer, cv2.IMREAD_COLOR)
-
-    def start_capture(self, **kwargs) -> MCastResponse:
-        """
-        :key client_identifier: str
-        """
-
-        client_identifier: str = get_kwarg(
-            kwargs=kwargs,
-            key="client_identifier",
-            arg_type=str)
-
-        if isinstance(self._capture_device_id, str) and self._capture_device_id.isnumeric():
-            self._capture_device_id.isnumeric = int(self._capture_device_id)
-        if self._capture is not None:
-            return EmptyResponse()
-        self.add_status_message(severity="info", message=f"{client_identifier} is starting capture.")
-
-        self._capture = self._detect_os_and_open_video(self._capture_device_id)
-        # NOTE: The USB3 cameras bought for this project appear to require some basic parameters to be set,
-        #       otherwise frame grab results in error
-        self._capture.set(cv2.CAP_PROP_FOURCC, cv2.VideoWriter_fourcc('M', 'J', 'P', 'G'))
-        self._capture.set(cv2.CAP_PROP_FRAME_WIDTH, 640)
-        self._capture.set(cv2.CAP_PROP_FRAME_HEIGHT, 480)
-        self._capture.set(cv2.CAP_PROP_FPS, 60)
-
-        self._capture_status.status = CaptureStatus.Status.RUNNING
-        return EmptyResponse()
-
-    def stop_capture(self, **kwargs) -> MCastResponse:
-        """
-        :key client_identifier: str
-        """
-
-        client_identifier: str = get_kwarg(
-            kwargs=kwargs,
-            key="client_identifier",
-            arg_type=str)
-
-        self.add_status_message(severity="info", message=f"{client_identifier} is stopping capture.")
-        if self._capture is not None:
-            self._capture.release()
-            self._capture = None
-        self._capture_status.status = CaptureStatus.Status.STOPPED
-        return EmptyResponse()
-
-    '''
     # Camera
+    def internal_update_capture(self):
+        msg = self._camera_interface.internal_update_capture()
+        if msg:
+            self.add_status_message(msg)
```

Comment:
I think severity might be omitted. But I also wonder if the overall status-reporting mechanism warrants a revisit. So I think it's okay to leave this as-is for now.